radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/P2PModule.java

```diff
@@ -0,0 +1,101 @@
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.network.p2p;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.multibindings.ProvidesIntoSet;
+import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.environment.EventProcessorOnRunner;
+import com.radixdlt.environment.LocalEvents;
+import com.radixdlt.environment.Runners;
+import com.radixdlt.network.hostip.HostIp;
+import com.radixdlt.network.p2p.PendingOutboundChannelsManager.PeerOutboundConnectionTimeout;
+import com.radixdlt.network.p2p.addressbook.AddressBookPersistence;
+import com.radixdlt.network.p2p.transport.PeerOutboundBootstrap;
+import com.radixdlt.network.p2p.transport.PeerOutboundBootstrapImpl;
+import com.radixdlt.properties.RuntimeProperties;
+import com.radixdlt.store.berkeley.BerkeleyAddressBookPersistence;
+
+public final class P2PModule extends AbstractModule {
+
+	private final RuntimeProperties properties;
+
+	public P2PModule(RuntimeProperties properties) {
+		this.properties = properties;
+	}
+
+	@Override
+	protected void configure() {
+		final var eventBinder = Multibinder.newSetBinder(binder(), new TypeLiteral<Class<?>>() { }, LocalEvents.class)
+			.permitDuplicates();
+		eventBinder.addBinding().toInstance(PeerEvent.class);
+		eventBinder.addBinding().toInstance(PeerOutboundConnectionTimeout.class);
+
+		bind(PeersView.class).to(PeerManagerPeersView.class);
+		bind(PeerOutboundBootstrap.class).to(PeerOutboundBootstrapImpl.class);
+		bind(AddressBookPersistence.class).to(BerkeleyAddressBookPersistence.class);
+	}
+
+	@ProvidesIntoSet
+	private EventProcessorOnRunner<?> peerManagerPeerEventProcessor(PeerManager peerManager) {
+		return new EventProcessorOnRunner<>(
+			Runners.P2P_NETWORK,
+			PeerEvent.class,
+			peerManager.peerEventProcessor()
+		);
+	}
+
+	@ProvidesIntoSet
+	private EventProcessorOnRunner<?> pendingOutboundChannelsManagerPeerEventProcessor(
+		PendingOutboundChannelsManager pendingOutboundChannelsManager
+	) {
+		return new EventProcessorOnRunner<>(
+			Runners.P2P_NETWORK,
+			PeerEvent.class,
+			pendingOutboundChannelsManager.peerEventProcessor()
+		);
+	}
+
+	@ProvidesIntoSet
+	private EventProcessorOnRunner<?> peerOutboundConnectionTimeoutEventProcessor(
+		PendingOutboundChannelsManager pendingOutboundChannelsManager
+	) {
+		return new EventProcessorOnRunner<>(
+			Runners.P2P_NETWORK,
+			PeerOutboundConnectionTimeout.class,
+			pendingOutboundChannelsManager.peerOutboundConnectionTimeoutEventProcessor()
+		);
+	}
+
+	@Provides
+	public P2PConfig p2pConfig() {
+		return P2PConfig.fromRuntimeProperties(this.properties);
+	}
+
+	@Provides
+	@Self
+	public RadixNodeUri selfUri(@Self ECPublicKey selfKey, HostIp hostIp, P2PConfig p2pConfig) {
```

Comment:
Not a fan of having to provide a whole `P2PConfig` to just get the port which is why I like annotations to single values better. But this is fine...for now.